src/agent/Core/ApplicationPool/Socket.h

```diff
@@ -69,6 +69,18 @@ struct Connection {
 			P_LOG_FILE_DESCRIPTOR_CLOSE(fd2);
 		}
 	}
+
+	void wait(int timeout = -1) {
+		boost::this_thread::disable_interruption di;
+        boost::this_thread::disable_syscall_interruption dsi;
+		TRACE_POINT();
+		pollfd pfd;
+		pfd.fd = fd;
+		pfd.events = POLLIN | POLLOUT;
+		if (-1 == syscalls::poll(&pfd, 1, timeout)) {
+			throw SystemException("poll() failed", errno);
```

Comment:
When throwing a SystemException, always first assign errno to a temporary variable. Otherwise the creation of the std::string for the exception might overwrite errno.